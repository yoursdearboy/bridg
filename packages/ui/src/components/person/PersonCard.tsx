import { InfoRow } from "@/components/InfoRow";
import {
  Badge,
  Box,
  Button,
  Card,
  Group,
  Modal,
  Stack,
  Text,
} from "@mantine/core";
import type { ApiPersonPerson } from "api-ts";
import dayjs from "dayjs";
import { useTranslation } from "react-i18next";
import { useDisclosure } from "@mantine/hooks";
import { EditPersonForm } from "./EditPersonForm";

interface PersonCardProps {
  person: ApiPersonPerson;
}

export const PersonCard = ({ person }: PersonCardProps) => {
  const { t } = useTranslation();
  const [opened, { open, close }] = useDisclosure(false);

  return (
    <>
      <Card withBorder shadow="sm" radius="md" padding="xs">
        <Card.Section withBorder inheritPadding py="xs">
          <Group justify="space-between">
            <Text fw={500} px="xs">
              {t("PersonShowPage.title")}
            </Text>
            <Button variant="outline" size="compact-sm" onClick={open} fw={500}>
              {t("PersonShowPage.edit")}
            </Button>
          </Group>
        </Card.Section>
        <Box pt="xs">
          <Stack gap="sm">
            <InfoRow
              label={t("Person.primaryName")}
              value={person.primaryName?.label}
            >
              {person.deathIndicator && (
                <Badge color="red" ml="sm">
                  {t("PersonCard.deceased")}
                </Badge>
              )}
            </InfoRow>

            <InfoRow
              label={t("Person.administrativeGenderCode")}
              value={
                person.administrativeGenderCode
                  ? t(`Gender.${person.administrativeGenderCode}`)
                  : t("na")
              }
            />

            <InfoRow
              label={t("Person.birthDate")}
              value={
                person.birthDate
                  ? t("intlDateTime", { val: person.birthDate })
                  : t("na")
              }
            />

            <InfoRow
              label={t("Person.age")}
              value={
                person.birthDate
                  ? t("dayjsDuration", {
                      val: dayjs.duration(
                        dayjs().diff(person.birthDate, "year"),
                        "year"
                      ),
                    })
                  : t("na")
              }
            />

            <InfoRow
              label={t("Person.deathDate")}
              value={
                person.deathIndicator
                  ? person.deathDate
                    ? t("intlDateTime", { val: person.deathDate })
                    : t("PersonCard.dateNotSpecified")
                  : t("PersonCard.notDeceased")
              }
            />
          </Stack>
        </Box>
      </Card>
      <Modal opened={opened} onClose={close} title={t("add")} size="lg">
        <EditPersonForm
          personId={person.id}
          person={person}
          onCancel={close}
          onSuccess={() => close()}
        />
      </Modal>
    </>
  );
};
