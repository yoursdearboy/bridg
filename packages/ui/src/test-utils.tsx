/* eslint-disable react-refresh/only-export-components */
import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  type AnyRoute,
  createRootRoute,
  createRootRouteWithContext,
  createRoute,
  createRouter,
  RouterProvider,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";
import type { PropsWithChildren } from "react";

const queryClient = new QueryClient();

const App = ({ children }: React.PropsWithChildren) => {
  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params, props }: { params?: object; props?: PropsWithChildren } = {}
) => {
  const rootRoute = createRootRouteWithContext()();
  let routeConfig = {
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  };
  if (props?.children)
    routeConfig = { ...routeConfig, component: () => props.children };

  const testRoute = createRoute(routeConfig);
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
    context: {
      queryClient,
    },
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};

export const renderComponent = (component: React.ReactNode) =>
  render(<App>{component}</App>);

export const EmptyRouterProvider = (props: PropsWithChildren) => {
  const rootRoute = createRootRoute({
    component: () => props.children,
  });

  return renderRoute(rootRoute, { props: { children: props.children } });
};
