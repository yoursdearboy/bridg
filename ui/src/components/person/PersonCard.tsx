import { InfoRow } from "@/components/InfoRow";
import { Badge, Card, Stack } from "@mantine/core";
import type { PersonOutput } from "bridg-ts";
import dayjs from "dayjs";
import { useTranslation } from "react-i18next";

interface PersonCardProps {
  person: PersonOutput;
}

export const PersonCard = ({ person }: PersonCardProps) => {
  const { t } = useTranslation();

  return (
    <Card withBorder shadow="sm" padding="lg" radius="md">
      <Stack gap="sm">
        <InfoRow label="Full Name" value={person.primaryName}>
          {person.deathIndicator && (
            <Badge color="red" ml="sm">
              {"Deceased"}
            </Badge>
          )}
        </InfoRow>

        <InfoRow label="Gender Name" value={person.administrativeGenderCode} />

        <InfoRow
          label="Date of Birth"
          value={
            person.birthDate
              ? t("intlDateTime", { val: person.birthDate })
              : undefined
          }
        />

        <InfoRow
          label="Age"
          value={
            person.birthDate
              ? t("dayjsDuration", {
                  val: dayjs.duration(
                    dayjs().diff(person.birthDate, "year"),
                    "year"
                  ),
                })
              : undefined
          }
        />

        <InfoRow
          label="Date of Death"
          value={
            person.deathIndicator
              ? person.deathDate
                ? t("intlDateTime", { val: person.deathDate })
                : "Date not specified"
              : "Not deceased"
          }
        />
      </Stack>
    </Card>
  );
};
