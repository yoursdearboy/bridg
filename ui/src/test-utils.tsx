import { MantineProvider } from "@mantine/core";
import { DatesProvider } from "@mantine/dates";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import {
  createRootRoute,
  createRoute,
  createRouter,
  RouterProvider,
  type AnyRoute,
} from "@tanstack/react-router";
import { act, render } from "@testing-library/react";
import "./i18n";

// eslint-disable-next-line react-refresh/only-export-components
const App = ({ children }: React.PropsWithChildren) => {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <MantineProvider defaultColorScheme="auto">
        <DatesProvider settings={{ locale: "en" }}>{children}</DatesProvider>
      </MantineProvider>
    </QueryClientProvider>
  );
};

export const renderRoute = (
  r: AnyRoute,
  { params }: { params?: object } = {}
) => {
  const rootRoute = createRootRoute();
  const testRoute = createRoute({
    ...r.options,
    params: params,
    getParentRoute: () => rootRoute,
    path: "/",
  });
  const routeTree = rootRoute.addChildren([testRoute]);
  const router = createRouter({
    routeTree,
  });
  return act(() =>
    render(
      <App>
        <RouterProvider router={router} />
      </App>
    )
  );
};
