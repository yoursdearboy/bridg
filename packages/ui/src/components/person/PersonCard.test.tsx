import { describe, expect, it } from "vitest";
import { EmptyRouterProvider, renderComponent } from "@/test-utils";
import { PersonCard } from "./PersonCard";
import { act } from "react";
import { render } from "@testing-library/react";

describe("PersonCard", () => {
  it("matches snapshot", async () => {
    const view = await act(() =>
      render(
        <EmptyRouterProvider>
          <PersonCard
            person={{
              administrativeGenderCode: "M",
              birthDate: new Date("2025-09-03"),
              deathDate: new Date("2025-09-03"),
              deathDateEstimatedIndicator: true,
              deathIndicator: true,
              id: "3fa85f64-5717-4562-b3fc-2c963f66afa6",
              primaryName: {
                use: "string",
                family: "string",
                given: "string",
                middle: "string",
                patronymic: "string",
                prefix: "string",
                suffix: "string",
                id: "3fa85f64-5717-4562-b3fc-2c963f66afa6",
                label: "string",
              },
            }}
          />
        </EmptyRouterProvider>
      )
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
});
