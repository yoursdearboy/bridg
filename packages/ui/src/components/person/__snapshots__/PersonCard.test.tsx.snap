// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PersonCard > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    data-with-border="true"
    style="--paper-radius: var(--mantine-radius-md); --paper-shadow: var(--mantine-shadow-sm); --card-padding: var(--mantine-spacing-xs);"
  >
    <div
      class="m_599a2148 mantine-Card-section"
      data-first-section="true"
      data-inherit-padding="true"
      data-with-border="true"
      style="padding-block: var(--mantine-spacing-xs);"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
      >
        <p
          class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          style="padding-inline: var(--mantine-spacing-xs); font-weight: 500;"
        >
          Person information
        </p>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-sm"
          data-variant="outline"
          style="--button-height: var(--button-height-compact-sm); --button-padding-x: var(--button-padding-x-compact-sm); --button-fz: var(--mantine-font-size-sm); --button-bg: transparent; --button-hover: var(--mantine-color-blue-outline-hover); --button-color: var(--mantine-color-blue-outline); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-blue-outline); font-weight: 500;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Edit person
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class=""
      style="padding-top: var(--mantine-spacing-xs);"
    >
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-sm); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: flex-start; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
          >
            Primary name:
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            string
            <span
              class="m_347db0ec mantine-Badge-root"
              style="--badge-bg: var(--mantine-color-red-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-left: var(--mantine-spacing-sm);"
            >
              <span
                class="m_5add502a mantine-Badge-label"
              >
                Deceased
              </span>
            </span>
          </p>
        </div>
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: flex-start; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
          >
            Gender:
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            Male
          </p>
        </div>
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: flex-start; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
          >
            Birth date:
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            9&#x2F;3&#x2F;2025
          </p>
        </div>
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: flex-start; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
          >
            Age:
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            a few seconds
          </p>
        </div>
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: flex-start; --group-justify: flex-start; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: var(--mantine-color-dimmed); font-weight: 600; width: calc(9.375rem * var(--mantine-scale));"
          >
            Death date:
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            9&#x2F;3&#x2F;2025
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
